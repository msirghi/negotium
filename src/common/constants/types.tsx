import React from 'react';
import { OverridableComponent } from '@mui/types';
import { SvgIconTypeMap } from '@mui/material';

export type MainMenuItem = {
  title: string;
  Icon: OverridableComponent<SvgIconTypeMap<{}, 'svg'>> & { muiName: string };
};
