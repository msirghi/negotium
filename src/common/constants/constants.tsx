import { MainMenuItem, ProjectColor, SiteTheme } from './types';
import StarBorderIcon from '@mui/icons-material/StarBorder';
import InboxIcon from '@mui/icons-material/Inbox';
import DateRangeIcon from '@mui/icons-material/DateRange';
import DoneIcon from '@mui/icons-material/Done';
import { SnackbarOrigin } from 'notistack';
import colors from '../styles/colors';
import { Descendant } from 'slate';

export const MAIN_MENU_ITEMS: MainMenuItem[] = [
  {
    Icon: InboxIcon,
    title: 'pageTitles.inbox',
    route: '/home/inbox',
  },
  {
    Icon: StarBorderIcon,
    title: 'pageTitles.today',
    route: '/home/today',
  },
  {
    Icon: DateRangeIcon,
    title: 'pageTitles.upcoming',
    route: '/home/upcoming',
  },
  // {
  //   Icon: DoneIcon,
  //   title: 'Completed',
  //   route: '/home/completed',
  // },
];

export const PROJECT_NAME = 'Negotium';

export const BASE_API_URL = 'http://localhost:4000';

export const BASE_API_URL_V1 = 'https://negotium-api.herokuapp.com/api/v1';

export const MAX_PROJECT_LIST_COUNT = 5;

export const SNACKBAR_POSITIONS: { [key: string]: SnackbarOrigin } = {
  BOTTOM_CENTER: { horizontal: 'center', vertical: 'bottom' },
};

export const PROJECT_COLORS: ProjectColor[] = [
  {
    color: 'rgb(219, 64, 53)',
    name: 'Red',
  },
  {
    color: 'rgb(255, 153, 51)',
    name: 'Orange',
  },

  {
    color: 'rgb(250, 208, 0)',
    name: 'Yellow',
  },
];

export const MENTION_ARRAY_KEYWORDS = ['Today', 'Tomorrow'];

export const SUPPORTED_LANGUAGES = [
  {
    code: 'en',
    title: 'English',
  },
  {
    code: 'ru',
    title: 'Russian',
  },
];

export const HOME_VIEW_LIST = [
  {
    title: 'Inbox',
    url: '/home/inbox',
  },
  {
    title: 'Today',
    url: '/home/today',
  },
];

export const TIME_FORMATS = [
  {
    title: '13:00',
    key: '24h',
  },
  {
    title: '1pm',
    key: 'am/pm',
  },
];

export const tempAccountInfo = {
  id: '1',
  email: 'mihail.sirghi@gmail.com',
  name: 'Mihail',
};

export const pagesWithoutWrapper = ['/login', '/registration'];

export const passwordStatusColors = {
  tooWeak: 'red',
  weak: 'orange',
  medium: 'purple',
  strong: 'green',
};

export const siteThemes: SiteTheme[] = [
  {
    color: colors.primaries.lightBlue_1,
    label: 'Default',
    internalKey: 'default',
  },
  {
    color: colors.black,
    label: 'Noir',
    internalKey: 'noir',
  },
];

export const initialRichTextValue: Descendant[] = [
  {
    type: 'paragraph',
    children: [{ text: '' }],
  },
];

export const ACTIVATION_CODE_VAR_NAME = 'ac';
