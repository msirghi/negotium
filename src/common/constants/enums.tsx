export enum PASSWORD_STRENGTH_STATUS {
  TOO_WEAK = 'Too weak',
  WEAK = 'Weak',
  MEDIUM = 'Medium',
  STRONG = 'Strong',
}

export enum ROW_DIRECTION {
  ROW = 'row',
  COLUMN = 'column',
}

export enum ACCOUNT_SETTINGS_CHANGE_MODE {
  EMAIL = 'EMAIL',
  PASSWORD = 'PASSWORD',
}

export enum HttpMethod {
  DELETE = 'DELETE',
  GET = 'GET',
  HEAD = 'HEAD',
  OPTIONS = 'OPTIONS',
  PATCH = 'PATCH',
  POST = 'POST',
  PUT = 'PUT',
}
