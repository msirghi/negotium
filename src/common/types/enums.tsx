export enum SETTINGS_OPTIONS {
    ACCOUNT = 'ACCOUNT',
    GENERAL = 'GENERAL',
    THEMES = 'THEMES'
}
