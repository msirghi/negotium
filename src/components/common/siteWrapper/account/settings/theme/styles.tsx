import { makeStyles } from '@mui/styles';

export const useThemeSettingsStyles = makeStyles({
  title: {
    fontSize: 14,
    padding: '12px 0'
  },
});
