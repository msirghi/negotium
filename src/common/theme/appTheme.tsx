// @ts-nocheck
import { createTheme } from '@mui/material/styles';
import colors from '../styles/colors';

const defaultFontFamily = `"Open Sans", "Roboto", "Helvetica", "Arial", sans-serif`;

export const defaultLightTheme = createTheme({
  typography: {
    allVariants: {
      fontFamily: defaultFontFamily,
    },
  },
  palette: {
    primary: {
      main: colors.primaries.lightBlue,
      light: colors.primaries.lightBlue_1,
      dark: colors.primaries.lightBlue,
    },
    custom: {
      menuIconBackground: colors.primaries.lightBlue,
      headerSearchBackground: colors.blues.light,
    },
  },
});

export const noirAppTheme = createTheme({
  typography: {
    allVariants: {
      fontFamily: defaultFontFamily,
    },
  },
  palette: {
    primary: {
      main: colors.greys['900'],
      light: colors.greys['800'],
      dark: colors.primaries.teal,
    },
    custom: {
      menuIconBackground: colors.greys['800'],
      headerSearchBackground: colors.greys['800'],
    }
  },
});

export const themeMap: { [key: string]: any } = {
  default: defaultLightTheme,
  noir: noirAppTheme,
};
