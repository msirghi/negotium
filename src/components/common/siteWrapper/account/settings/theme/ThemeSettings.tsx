import { ThemeItem } from './item/ThemeItem';
import { siteThemes } from '../../../../../../common/constants/constants';
import { Grid } from '@mui/material';
import { useThemeSettingsStyles } from './styles';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from '../../../../../../redux/store';
import { setSiteTheme } from '../../../../../../redux/account/accountSlice';
import AccountService from '../../../../../../services/AccountService';

export const ThemeSettings = () => {
  const classes = useThemeSettingsStyles();
  const siteTheme = useSelector(
    (state: RootState) => state.account.metadata.theme
  );
  const dispatch = useDispatch();

  const onThemeChange = async (theme: string) => {
    if (siteTheme !== theme) {
      dispatch(setSiteTheme(theme));
      await AccountService.updateUserTheme(theme);
    }
  };

  return (
    <div>
      <div className={classes.title}>
        Choose a theme that matches your preference.
      </div>
      <Grid container spacing={3}>
        {siteThemes.map((theme) => (
          <Grid key={theme.color} item>
            <ThemeItem
              theme={theme}
              selected={siteTheme === theme.internalKey}
              onClick={onThemeChange}
            />
          </Grid>
        ))}
      </Grid>
    </div>
  );
};
