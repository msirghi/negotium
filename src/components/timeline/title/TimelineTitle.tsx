import { makeStyles } from '@mui/styles';
import { FC } from 'react';
import { Task } from '../../../common/types/tasks.types';
import { TaskDateType } from '../../../common/constants/enums';
import { useTheme } from '@mui/system';
import colors from '../../../common/styles/colors';

const useStyles = makeStyles({
  title: (props: { color: string }) => ({
    fontSize: 16,
    color: props.color,
  }),
});

type Props = {
  title: string;
  dateType: Task['dueDate'];
};

export const TimelineTitle: FC<Props> = ({ title, dateType }) => {
  const isInThePast = dateType === TaskDateType.PAST;
  const theme = useTheme();
  const classes = useStyles({ color: isInThePast ? theme.palette.error.main : colors.black });

  return <p className={classes.title}>{title}</p>;
};
