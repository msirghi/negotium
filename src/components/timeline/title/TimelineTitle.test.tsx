import { TaskDateType } from '../../../common/constants/enums';
import { ThemeProvider } from '@mui/system';
import { noirAppTheme } from '../../../common/theme/appTheme';
import { TimelineTitle } from './TimelineTitle';
import renderer from 'react-test-renderer';

describe('TimelineTitle', () => {
  const defaultProps = {
    title: 'Title',
    dateType: TaskDateType.Today,
  };

  const renderComponent = () => {
    return (
      <ThemeProvider theme={noirAppTheme}>
        <TimelineTitle {...defaultProps} />
      </ThemeProvider>
    );
  };

  it('should match the snapshot', () => {
    const wrapper = renderer.create(renderComponent());
    expect(wrapper).toMatchSnapshot();
  });
});
