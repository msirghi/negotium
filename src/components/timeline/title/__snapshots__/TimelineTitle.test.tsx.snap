// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineTitle should match the snapshot 1`] = `
<p
  className="makeStyles-title-1 makeStyles-title-2"
>
  Title
</p>
`;
